cats/__init__.py

```diff
@@ -131,38 +240,32 @@ def main(arguments=None):
     now_avg, best_avg = get_avg_estimates(
         CI_forecast, duration=duration
     )
-    sys.stderr.write(str(best_avg) + "\n")
-
-    sys.stderr.write(f"Best job start time: {best_avg.start}\n")
-    print(f"{best_avg.start:%Y%m%d%H%M}")  # for POSIX compatibility with at -t
+    output = CATSOutput(choice_CI_API, now_avg, best_avg, location, "GBR")
 
     ################################
     ## Calculate carbon footprint ##
     ################################
 
-    error_message = "Not enough information to estimate total carbon footprint, both --jobinfo and config files are needed.\n"
-
     if args.jobinfo:
         jobinfo = validate_jobinfo(args.jobinfo, expected_partition_names=config['partitions'].keys())
 
         if not (jobinfo and config):
-            sys.stderr.write(error_message)
+            logging.warning("Not enough information to estimate total carbon footprint, "
+                            "both --jobinfo and config files are needed.\n")
         else:
-            estim = greenAlgorithmsCalculator(
+            output.emmissionEstimate = greenAlgorithmsCalculator(
                 config=config,
                 runtime=timedelta(minutes=args.duration),
                 averageBest_carbonIntensity=best_avg.value, # TODO replace with real carbon intensity
                 averageNow_carbonIntensity=now_avg.value,
                 **jobinfo,
             ).get_footprint()
-
-            sys.stderr.write(f"Estimated emmissions for running job now: {estim.now}\n")
-            msg = (
-                f"Estimated emmissions for running delayed job: {estim.best})\n"
-                f" (- {estim.savings})"
-            )
-            sys.stderr.write(msg)
-
+    if args.format == "json":
+        print(output.to_json(dateformat=args.dateformat or args.scheduler, sort_keys=True, indent=2))
+    else:
+        print(output)
+    if args.command and args.scheduler == "at":
+        schedule_at(output, args.command.split())
```

Comment:
```
$ cats -d 5 --loc OX4 -c 'ls'
Best job start time: 2024-03-05 00:15:20.609896+00:00
```
I'd expect `cats` to tell me off that I haven't specified a scheduler instead of just returning the start time. I was left wondering whether or not `ls` had actually been scheduled :)

on the other hand I think it's okay that `cats -d 5 --loc OX4 --scheduler "at"` basically ignores the `--scheduler` option just returns the job start time, because the user didn't specify any command to be run anyway.

Reply:
Thanks, good catch!